de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inventory/DraftInventoryLinesCreator.java

```diff
@@ -0,0 +1,152 @@
+package de.metas.handlingunits.inventory;
+
+import static org.adempiere.model.InterfaceWrapperHelper.saveRecord;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.adempiere.util.Check;
+import org.adempiere.util.GuavaCollectors;
+import org.adempiere.util.Services;
+import org.compiere.model.I_M_Inventory;
+
+import de.metas.document.engine.IDocumentBL;
+import de.metas.handlingunits.HuId;
+import de.metas.handlingunits.IHandlingUnitsBL;
+import de.metas.handlingunits.model.I_M_HU;
+import de.metas.handlingunits.model.I_M_InventoryLine;
+import de.metas.handlingunits.storage.IHUProductStorage;
+import de.metas.inventory.IInventoryDAO;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+import lombok.experimental.NonFinal;
+
+/*
+ * #%L
+ * de.metas.handlingunits.base
+ * %%
+ * Copyright (C) 2018 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+/**
+ * Creates or updates inventory lines for
+ *
+ * @author metas-dev <dev@metasfresh.com>
+ *
+ */
+@Value
+public class DraftInventoryLinesCreator
+{
+	IDocumentBL documentBL = Services.get(IDocumentBL.class);
+	IInventoryDAO inventoryDAO = Services.get(IInventoryDAO.class);
+	IHandlingUnitsBL handlingUnitsBL = Services.get(IHandlingUnitsBL.class);
+
+	I_M_Inventory inventoryRecord;
+
+	Map<HuId, I_M_InventoryLine> inventoryLinesByHU;
+	final Set<Integer> seenLocatorIs = new HashSet<>();
+
+	@NonFinal
+	long countInventoryLines = 0;
+
+	@Builder
+	private DraftInventoryLinesCreator(
```

Comment:
please don't call other methods from ctor.
Simply create an `execute()` method and move all this code there.
See https://en.wikipedia.org/wiki/Command_pattern